packages/cw721-controllers/src/nft_claim.rs

```diff
@@ -0,0 +1,422 @@
+use cosmwasm_std::{Addr, BlockInfo, CustomQuery, Deps, StdResult, Storage};
+use cw_storage_plus::Map;
+use cw_utils::Expiration;
+use schemars::JsonSchema;
+use serde::{Deserialize, Serialize};
+
+#[derive(Serialize, Deserialize, Clone, Debug, PartialEq, JsonSchema)]
```

Comment:
does this file need to be in packages because it will also be used by another contract other than `stake-cw721`?

Reply:
That's the idea. 🤷‍♂️ In theory this has a really similar setup to cw-controllers in cw-plus which is used for a variety of things.